nodeup/pkg/model/kube_apiserver.go

```diff
@@ -312,6 +312,13 @@ func (b *KubeAPIServerBuilder) buildPod() (*v1.Pod, error) {
 		}
 	}
 
+	hasEventsEtcdCluster := false
+	for _, etcdCluster := range b.Cluster.Spec.EtcdClusters {
+		if etcdCluster.Name == "events" {
+			hasEventsEtcdCluster = true
+		}
+	}
```

Comment:
Trying to explain my previous comment, nodeup section:
```suggestion
	if !b.IsMaster {
		clusterName := b.Cluster.ObjectMeta.Name
		mainEtcdDNSName := "main.etcd." + clusterName
		eventsEtcdDNSName := "events.etcd." + clusterName
		for i := range kubeAPIServer.EtcdServers {
			kubeAPIServer.EtcdServers[i] = strings.ReplaceAll(kubeAPIServer.EtcdServers[i], "127.0.0.1", mainEtcdDNSName)
		}
		for i := range kubeAPIServer.EtcdServersOverrides {
			if strings.HasPrefix(kubeAPIServer.EtcdServersOverrides[i], "/events") {
				kubeAPIServer.EtcdServersOverrides[i] = strings.ReplaceAll(kubeAPIServer.EtcdServersOverrides[i], "127.0.0.1", eventsEtcdDNSName)
			}
		}
	}
```